src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -78,6 +78,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
 
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
+
+    if(username.length()<3 || username.length()>14){
+        request.setAttribute("error", "username must be between 3 and 14 characters");
+        request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+        return;
+    }
+
+    if(password.length()<5||password.length()>13 || !password.matches("(.*[a-z].*)")|| !password.matches( "(.*[0-9].*)")){
```

Comment:
Don't we already have this same check in NewPasswordServlet? The same logic like this shouldn't exist in 2 places...
What about doing something like this. Make a new class called say PasswordUtils, with a static method like isValidPassword() returning boolean, that you'd then call both here and in NewPasswordServlet? 
You could of course put a such a method in one of the 2 servlet classes as well, but I think that would detract from their cohesiveness, this is really a "business logic" sort of function which should be decoupled from the servlets if it is used in more than one place. 
WDYT?

Reply:
really sorry for late response, my professor gave me a lot to do this week with the research i'm doing and I spent some time practicing for the mock interview as well, for this class though will the only method/s be for incorrect passwords?

Reply:
My point is we shouldn't have non-trivial logic like this in more than one place.. if we need it in multiple places(ie in both RegisterServlet and NewPasswordServlet) then we should put it in a some sort of common utility class that both servlets can use. Then if we decide to change the password rules we only have to do it in one place. Does that make sense?